### PR TITLE
docs: add changelog about redis cache

### PR DESCRIPTION
**Pull request by hanjm**
* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

1. docs: add changelog about redis cache.
2. fix code review in #4888.

### CODE REVIEW

File: pkg/queryfrontend/cache.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func (t thanosCacheKeyGenerator) GenerateCacheKey(userID string, r queryrange.Re
 		i := 0
 		for ; i < len(t.resolutions) && t.resolutions[i] > tr.MaxSourceResolution; i++ {
 		}
-		// Cache key should has a uniq prefix, We use `fe` represent frontend.
+		// Cache key should have a unique prefix. `fe` represents frontend.
```
Comment (yeya24) [inline pkg/queryfrontend/cache.go:37]: Why do we need to mention `fe` here? Can we remove it?

**Review by yeya24 — APPROVED** [review]
LGTM!

File: pkg/queryfrontend/cache.go (hunk 1 above)
Comment (GiedriusS) [reply pkg/queryfrontend/cache.go:37]: @hanjm let's just remove this comment (:

Comment (hanjm) [reply pkg/queryfrontend/cache.go]: @GiedriusS Thanks. Done.

**Review by GiedriusS — APPROVED** [review]

**Pull request merged**